### DISCUSSION TITLE
How can I use oxc extension in vscode?

### DISCUSSION THREAD

**Discussion by markEHVN** [oxc-project/oxc#7258] [Q&A] [answered]
## Description
- I installed oxc extension in vscode, I added `.oxlintrc.json` file. 

When I saw settings of oxc extension. To be honest, I do not know what do settings mean?. I try to change some settings but it does not active. Can some body used oxc extension? What is your settings? Can you give me an repo to try oxc's settings? 
Thank you so much ❤️.

ID | Description | Default
-- | -- | --
oxc.configPath | Path to ESlint configuration. | ".eslintrc"
oxc.enable | enable oxc language server | true
oxc.lint.run | Run the linter on save (onSave) or on type (onType) | "onType"
oxc.path.server | Path to Oxc language server binary. | ""
oxc.trace.server | Traces the communication between VS Code and the language server. | "off"

## Some problems that I have been seen
1. Settings's extension have been not apply.
2. Autofix rules have been not fixed with auto save (I can fix it from cli).
3. We do not have a documentation.

**Comment by markEHVN** [comment]
I try change `oxc.configPath = .oxlintrc.json`, changed `oxc.trace.server = "verbose"`,  changed `oxc.enable = false`, after reverted `oxc.enable=true` but Oxc output does not correctly work. 
Sometime, when I try to change `oxc.enable`, oxc output also does not correctly work.

**Comment by nrayburn-tech** [comment, marked as answer]
When changing a setting, you need to restart your VS Code application. This is something that can be fixed from the extension but it’s still early and being worked on actively. This probably relates to item 1.

Item 2 isn’t supported yet (search for an issue, if one doesn’t exist for it then feel free to make one).

Item 3 is definitely something that can be improved on. I wrote up a short summary of some features recently, might be useful here. Still need an enhancement to the actual extension documentation though.
https://github.com/oxc-project/oxc/issues/2719#issuecomment-2466990008

**Reply by Enrypase** [reply]
Possibly, could you share your settings  that allow the extension to work?
Do you rely still on ESLint? If yes, for which reason?

**Reply by husayt** [reply]
this was asked over a year ago, but I have to switch back to eslint in vscode, as oxc wouldn't work. Speed is all cool, but ide support is key and if it doesn't work in ide I am crippled. Eslint bit slower, but at least works

**Reply by nrayburn-tech** [reply]
There’s definitely nothing wrong with that and I encourage you to use what works for you.

Oxlint is still being actively worked on to deliver features that ESLint already has. If you need those missing features, then you should keep using ESLint. However, if you don’t need those features or can justify the complexity of running both tools within your workflow then the performance from Oxlint may be beneficial depending on your use case.

That being said, Oxlint isn’t near as mature as ESLint. It’s going to take time to continue polishing IDE support and keeping the IDE support updated as Oxlint delivers new features.
